src/com/TTT/TurnUI.java

```diff
@@ -1,14 +1,47 @@
 package com.TTT;
 
 import java.io.InputStream;
+import java.io.PrintStream;
 import java.util.Scanner;
 
 public class TurnUI {
 
-    public void takeTurn(Board board, BoardPrinter printer, Game game, InputStream in) {
+    private PrintStream out;
+    private BoardPrinter printer;
+    private Scanner scanner;
+    private String inputPrompt ="Please enter a space number: ";
+
+    public TurnUI(BoardPrinter printer, PrintStream out, InputStream in) {
+        this.printer = printer;
+        this.out = out;
+        scanner = new Scanner(in);
+        scanner.useDelimiter("\n");
+    }
+
+    public void takeTurn(Board board, Game game) {
+        int selectedSpace = getPlayerInput(board, inputPrompt);
+
+        while(selectedSpace < 0 || selectedSpace > (board.size() - 1) || board.isOccupied(selectedSpace))
+        {
```

Comment:
I'm looking for consistency in formatting. K&R style would have the opening / close braces for class / method at the same indentation. I don't think you should drop the brace to the next line within a method. I would personally like to see the brace on the same line in all cases so there's no difference.